### PR TITLE
Feature/another iterator iteration

### PR DESCRIPTION
**Pull request by trasher**

### CODE REVIEW

File: report/computer_last_inventory.php [hunk 1]
```diff
@@ -92,21 +92,36 @@
...
+        'OR' => [
+            new \QueryExpression("NOW() > ADDDATE(last_inventory_update, INTERVAL " . $nbdays . " DAY"),
+            ['last_inventory_update' => null]
+        ]
+    ] + $state_where + getEntitiesRestrictCriteria('glpi_computers'),
+    'ORDER' => [
+        'last_inventory_update' => 'DESC'
+    ]
+]);
+$result = $iterator->next();
```
Comment (cedric-anne) [inline report/computer_last_inventory.php:124]: This is not correct. Indeed, `$result` is used on following lines in `$DB->numrows($result)` and `while ($data = $DB->fetchArray($result)) {`.

Comment (trasher) [reply report/computer_last_inventory.php:124]: Indeed, fixed

File: report/computer_last_inventory.php [hunk 2]
```diff
@@ -92,26 +92,41 @@
...
```
Comment (cedric-anne) [inline report/computer_last_inventory.php:124]: This is useless now.
```suggestion
```

File: report/switch_ports.history.php [hunk 3]
```diff
@@ -51,16 +51,32 @@
...
-          WHERE `itemtype`='NetworkEquipment'
-          ORDER BY `glpi_networkequipments`.`name`, `glpi_networkports`.`logical_number`;";
+$iterator = $DB->request([
+    'SELECT' => [
+        'glpi_networkequipments.name AS name',
+        'glpi_networkports.name AS pname',
+        'glpi_networkports.id AS id'
+    ],
+    'FROM'   => 'glpi_networkequipments',
+    'JOIN'   => [
```
Comment (g-bougard) [inline report/switch_ports.history.php:61]: There was a `LEFT JOIN` in the original query. As I'm not a sql join expert, can you confirm a simple `JOIN` is equivalent here or if this is a mistake ?

Comment (trasher) [reply report/switch_ports.history.php:61]: `JOIN` is indeed a shortcut for `LEFT JOIN`, but I do not like to use it; therefore it's still a mistake :)
```suggestion
    'LEFT JOIN'   => [
```

**Review by g-bougard — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**